cpp/src/Slice/SliceUtil.cpp

```diff
@@ -656,9 +656,39 @@ Slice::unwrapIfOptional(const TypePtr& type)
     {
         return optional->underlying();
     }
-    else
+    return type;
+}
+
+TypePtr
+Slice::unwrapIfAlias(const TypePtr& type)
+{
+    if (auto typealias = TypeAliasPtr::dynamicCast(type))
     {
-        return type;
+        return typealias->underlying();
+    }
+    return type;
+}
+
+TypePtr
+Slice::unwrapType(const TypePtr& type)
+{
+    return unwrapIfOptional(unwrapIfAlias(type));
+}
+
+void
+Slice::resolveAlias(TypePtr& type, StringList& metadata)
```

Comment:
I find the signature of this resolve function with 2 in-out parameters inelegant.

```
(TypePtr, StringList) resolveAlias(const TypePtr type&, const StringList& metadata);
```
looks better.

Reply:
Normally, I'd agree that in-out parameters are 'more risky', but everywhere we use this method, we want to directly overwrite something in place. Making the parameters const references and instead returning the outputs would just be adding a pointless step to this, since we're going to overwrite the original values anyways.

And while you think it makes the function signature look nicer, it would make the implementation and everywhere where we use it worse. For instance, this:
```
resolveAlias(taggedDef->type, taggedDef->metadata);
```
would become this:
```
tie(taggedDef->type, taggedDef->metadata) = resolveAlias(taggedDef->type, taggedDef->metadata);
```
We can't use structured binding, because you can only bind to new variables, not existing ones, like we have to here.

The other advantage of not returning anything is that when the provided type isn't an alias (which is often), the method can just immediately return and nothing is changed.
If we used the tuple return type like you suggest, then the method would instead return an exact copy of the inputs, and in the grammar we would overwrite the type and metadata with an exact copy of themselves.

Reply:
Then I suggest to change the name of the function to for example `unalias`.

Reply:
`unalias` works for me. I changed it.